kotest-framework/kotest-framework-engine/src/jvmMain/kotlin/io/kotest/engine/test/DuplicateTestNameHandler.kt

```diff
@@ -0,0 +1,35 @@
+package io.kotest.engine.test
+
+import io.kotest.core.test.DescriptionName
+import io.kotest.core.test.DuplicateTestNameMode
+import io.kotest.core.test.Identifiers
+
+/**
+ * A handler for test names. Will track all test names used for a particular context.
+ */
+class DuplicateTestNameHandler(private val mode: DuplicateTestNameMode) {
+
+   private val names = mutableSetOf<String>()
+
+   private fun message(name: String): String =
+      "Duplicated test name ${name}. To disable this message, set DuplicateTestNameMode to None."
```

Comment:
Currently the warning suggests to use "None" as value, however, "None" is not a valid enum value of DuplicateTestNameMethod. Instead of "None", "Silent" should be used in the warning message.

Reply:
This pr is pretty old.
I created #3069 to address this.